[Bug]: python-olm build fail

### Bug Description

during the update from ver0.5.0 to 0.6.0, error was shown. however it seems not impacting hermes agent behaviour.

⚕ Updating Hermes Agent...

→ Fetching updates...
→ Local changes detected — stashing before update...
Saved working directory and index state On main: hermes-update-autostash-20260331-050318
→ Found 30 new commit(s)
→ Pulling updates...

⚠ Local changes were stashed before updating.
  Restoring them may reapply local customizations onto the updated codebase.
  Review the result afterward if Hermes behaves unexpectedly.
Restore local changes now? [Y/n]
y
→ Restoring local changes...
⚠ Local changes were restored on top of the updated codebase.
  Review `git diff` / `git status` if Hermes behaves unexpectedly.
  ✓ Cleared 14 stale __pycache__ directories
→ Updating Python dependencies...
  × Failed to build `python-olm==3.2.16`
  ├─▶ The build backend returned an error
  ╰─▶ Call to `setuptools.build_meta.build_wheel` failed (exit status: 1)

      [stdout]
      -- Configuring incomplete, errors occurred!

      [stderr]
      CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
        Compatibility with CMake < 3.5 has been removed from CMake.

        Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
        to tell CMake that the project requires at least <min> but has been updated
        to work with policies introduced by <max> or earlier.

        Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.


      Traceback (most recent call last):
        File "<string>", line 14, in <module>
        File "/Users/kazamak/.cache/uv/builds-v0/.tmpGvFeVL/lib/python3.11/site-packages/setuptools/build_meta.py",
      line 333, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=[])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/Users/kazamak/.cache/uv/builds-v0/.tmpGvFeVL/lib/python3.11/site-packages/setuptools/build_meta.py",
      line 301, in _get_build_requires
          self.run_setup()
        File "/Users/kazamak/.cache/uv/builds-v0/.tmpGvFeVL/lib/python3.11/site-packages/setuptools/build_meta.py",
      line 317, in run_setup
          exec(code, locals())
        File "<string>", line 4, in <module>
        File "/Users/kazamak/.cache/uv/builds-v0/.tmpGvFeVL/lib/python3.11/site-packages/setuptools/__init__.py", line
      117, in setup
          return distutils.core.setup(**attrs)  # type: ignore[return-value]
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/Users/kazamak/.cache/uv/builds-v0/.tmpGvFeVL/lib/python3.11/site-packages/setuptools/_distutils/core.py",
      line 148, in setup
          _setup_distribution = dist = klass(attrs)
                                       ^^^^^^^^^^^^
        File "/Users/kazamak/.cache/uv/builds-v0/.tmpGvFeVL/lib/python3.11/site-packages/setuptools/dist.py", line 321,
      in __init__
          _Distribution.__init__(self, dist_attrs)
        File "/Users/kazamak/.cache/uv/builds-v0/.tmpGvFeVL/lib/python3.11/site-packages/setuptools/_distutils/dist.py",
      line 307, in __init__
          self.finalize_options()
        File "/Users/kazamak/.cache/uv/builds-v0/.tmpGvFeVL/lib/python3.11/site-packages/setuptools/dist.py", line 790,
      in finalize_options
          ep(self)
        File "/Users/kazamak/.cache/uv/builds-v0/.tmpGvFeVL/lib/python3.11/site-packages/setuptools/dist.py", line 810,
      in _finalize_setup_keywords
          ep.load()(self, ep.name, value)
        File "/Users/kazamak/.cache/uv/builds-v0/.tmpGvFeVL/lib/python3.11/site-packages/cffi/setuptools_ext.py", line
      229, in cffi_modules

### Steps to Reproduce

1. run "hermes update"


### Expected Behavior

build success

### Actual Behavior

Error displays

### Affected Component

Other

### Messaging Platform (if gateway-related)

N/A (CLI only)

### Operating System

mac os 

### Python Version

3.14.3

### Hermes Version

0.6.0

### Relevant Logs / Traceback

```shell
⚕ Updating Hermes Agent...

→ Fetching updates...
→ Local changes detected — stashing before update...
Saved working directory and index state On main: hermes-update-autostash-20260331-050318
→ Found 30 new commit(s)
→ Pulling updates...

⚠ Local changes were stashed before updating.
  Restoring them may reapply local customizations onto the updated codebase.
  Review the result afterward if Hermes behaves unexpectedly.
Restore local changes now? [Y/n]
y
→ Restoring local changes...
⚠ Local changes were restored on top of the updated codebase.
  Review `git diff` / `git status` if Hermes behaves unexpectedly.
  ✓ Cleared 14 stale __pycache__ directories
→ Updating Python dependencies...
  × Failed to build `python-olm==3.2.16`
  ├─▶ The build backend returned an error
  ╰─▶ Call to `setuptools.build_meta.build_wheel` failed (exit status: 1)

      [stdout]
      -- Configuring incomplete, errors occurred!

      [stderr]
      CMake Error at CMakeLists.txt:1 (cmake_minimum_required):
        Compatibility with CMake < 3.5 has been removed from CMake.

        Update the VERSION argument <min> value.  Or, use the <min>...<max> syntax
        to tell CMake that the project requires at least <min> but has been updated
        to work with policies introduced by <max> or earlier.

        Or, add -DCMAKE_POLICY_VERSION_MINIMUM=3.5 to try configuring anyway.


      Traceback (most recent call last):
        File "<string>", line 14, in <module>
        File "/Users/kazamak/.cache/uv/builds-v0/.tmpGvFeVL/lib/python3.11/site-packages/setuptools/build_meta.py",
      line 333, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=[])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/Users/kazamak/.cache/uv/builds-v0/.tmpGvFeVL/lib/python3.11/site-packages/setuptools/build_meta.py",
      line 301, in _get_build_requires
          self.run_setup()
        File "/Users/kazamak/.cache/uv/builds-v0/.tmpGvFeVL/lib/python3.11/site-packages/setuptools/build_meta.py",
      line 317, in run_setup
          exec(code, locals())
        File "<string>", line 4, in <module>
        File "/Users/kazamak/.cache/uv/builds-v0/.tmpGvFeVL/lib/python3.11/site-packages/setuptools/__init__.py", line
      117, in setup
          return distutils.core.setup(**attrs)  # type: ignore[return-value]
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/Users/kazamak/.cache/uv/builds-v0/.tmpGvFeVL/lib/python3.11/site-packages/setuptools/_distutils/core.py",
      line 148, in setup
          _setup_distribution = dist = klass(attrs)
                                       ^^^^^^^^^^^^
        File "/Users/kazamak/.cache/uv/builds-v0/.tmpGvFeVL/lib/python3.11/site-packages/setuptools/dist.py", line 321,
      in __init__
          _Distribution.__init__(self, dist_attrs)
        File "/Users/kazamak/.cache/uv/builds-v0/.tmpGvFeVL/lib/python3.11/site-packages/setuptools/_distutils/dist.py",
      line 307, in __init__
          self.finalize_options()
        File "/Users/kazamak/.cache/uv/builds-v0/.tmpGvFeVL/lib/python3.11/site-packages/setuptools/dist.py", line 790,
      in finalize_options
          ep(self)
        File "/Users/kazamak/.cache/uv/builds-v0/.tmpGvFeVL/lib/python3.11/site-packages/setuptools/dist.py", line 810,
      in _finalize_setup_keywords
          ep.load()(self, ep.name, value)
        File "/Users/kazamak/.cache/uv/builds-v0/.tmpGvFeVL/lib/python3.11/site-packages/cffi/setuptools_ext.py", line
      229, in cffi_modules
```

### Root Cause Analysis (optional)

_No response_

### Proposed Fix (optional)

_No response_

### Are you willing to submit a PR for this?

- [ ] I'd like to fix this myself and submit a PR

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -76,7 +76,10 @@ all = [
   "hermes-agent[modal]",
   "hermes-agent[daytona]",
   "hermes-agent[messaging]",
-  "hermes-agent[matrix]",
+  # matrix excluded: python-olm (required by matrix-nio[e2e]) is upstream-broken
+  # on modern macOS (archived libolm, C++ errors with Clang 21+). Including it
+  # here causes the entire [all] install to fail, dropping all other extras.
+  # Users who need Matrix can install manually: pip install 'hermes-agent[matrix]'
   "hermes-agent[cron]",
   "hermes-agent[cli]",
   "hermes-agent[dev]",

--- a/tests/test_project_metadata.py
+++ b/tests/test_project_metadata.py
@@ -11,8 +11,12 @@ def _load_optional_dependencies():
     return project["optional-dependencies"]
 
 
-def test_all_extra_includes_matrix_dependency():
+def test_matrix_extra_exists_but_excluded_from_all():
+    """matrix-nio[e2e] depends on python-olm which is upstream-broken on modern
+    macOS (archived libolm, C++ errors with Clang 21+).  The [matrix] extra is
+    kept for opt-in install but deliberately excluded from [all] so one broken
+    upstream dep doesn't nuke every other extra during ``hermes update``."""
     optional_dependencies = _load_optional_dependencies()
 
     assert "matrix" in optional_dependencies
-    assert "hermes-agent[matrix]" in optional_dependencies["all"]
+    assert "hermes-agent[matrix]" not in optional_dependencies["all"]

